otherlibs/unix/strofaddr.c

```diff
@@ -28,22 +28,22 @@ CAMLprim value caml_unix_string_of_inet_addr(value a)
 #ifdef HAS_IPV6
 #ifdef _WIN32
   char buffer[64];
-  union sock_addr_union sa;
+  struct sockaddr_storage addr;
   int len;
   int retcode;
   if (caml_string_length(a) == 16) {
-    memset(&sa.s_inet6, 0, sizeof(struct sockaddr_in6));
-    sa.s_inet6.sin6_family = AF_INET6;
-    sa.s_inet6.sin6_addr = GET_INET6_ADDR(a);
+    memset(&addr, 0, sizeof(struct sockaddr_storage));
```

Comment:
```suggestion
    memset(&addr, 0, sizeof(struct sockaddr_storage));
```
This line and the one in the other branch can now be factorised.

Reply:
Done! Note that the Windows code path can be completely eliminated by switching to `inet_ntop`/`InetNtop`. I'll do that in a follow-up.